### ISSUE TITLE
Bug with anonymous actions on infolist

### ISSUE THREAD

**Issue by sbc640964** [laravel-filament/filament#7630]
### Package

filament/filament

### Package Version

v3.0.15

### Laravel Version

v10.18.0

### PHP Version

PHP 8.1

### Problem description

I followed the documents
https://github.com/sbc640964/filament-infolists-action
This is my code
```php
public static function infolist(Infolist $infolist): Infolist
{
    return $infolist->schema([
        Infolists\Components\TextEntry::make('name'),
        Infolists\Components\Actions::make([
            Infolists\Components\Actions\Action::make('Edit')
                ->form(fn($form) => static::form($form))
        ]),
    ]);
}
```
I get an error open the infolist modal:
**Typed property Filament\Infolists\Infolist::$name must not be accessed before initialization**

### Expected behavior

Everything will go smoothly... :)

### Steps to reproduce

Installing Laravel
```cli
composer create-project laravel/laravel example-app
```
Installing Filament
```cli
composer require filament/filament:"^3.0-stable" -W
php artisan filament:install --panels
```
Adding Resource (simple)
```cli
php artisan make:filament-resource User --simple
```
Adding ViewAction to table actions
```php
public static function table(Table $table): Table
{
    return $table
        ....
        ->actions([
            ...
            Tables\Actions\ViewAction::make(),
        ])
       ...
}
```
Added anonymous action to infolist
```php
public static function infolist(Infolist $infolist): Infolist
{
    return $infolist->schema([
        Infolists\Components\TextEntry::make('name'),
        Infolists\Components\Actions::make([
            Infolists\Components\Actions\Action::make('Edit')
                ->form(fn($form) => static::form($form))
        ]),
    ]);
}
```

### Reproduction repository

https://github.com/sbc640964/filament-infolists-action

### Relevant log output

```shell
https://flareapp.io/share/LPdkYjXm
```

**Comment by mikecpl** [comment]
Setting the name of the infolist will solve your problem:
```
public static function infolist(Infolist $infolist): Infolist
{
    return $infolist
        ->name('something')
        ->schema([
            Infolists\Components\TextEntry::make('name'),
            Infolists\Components\Actions::make([
                Infolists\Components\Actions\Action::make('Edit')
                    ->form(fn($form) => static::form($form))
            ]),
        ]);
}
```

But after that I am getting a different error:
```
Unable to call component method. Public method [mountInfolistAction] not found on component
```
I am trying to figure it out what could be the solution

**Comment by gabtorr** [comment]
Same here.  In my scenario, rather than having an anonymous action, the action was appended as a suffixAction to a TextEntry component inside a RepeteableEntry.

**Comment by breakzhuo** [comment]
> Same here. In my scenario, rather than having an anonymous action, the action was appended as a suffixAction to a TextEntry component inside a RepeteableEntry.

Same too.

**Comment by commongoodguy** [comment]
```
 Unable to call component method. Public method [mountInfolistAction] not found on component
 ```

In my case this error is caused by not importing/using the trait `Filament\Infolists\Concerns\InteractsWithInfolists`, remember to also implement `Filament\Infolists\Contracts\HasInfolists`.

They're not imported by default by `Filament\Pages\BasePage` and of course you have to import them manually on a custom Livewire component.

**Comment by nprasath002** [comment]
I am trying to add an anonymous action button that will slideover a form and getting the same error as above

```
    public static function infolist(Infolist $infolist): Infolist
    {
        return $infolist
            ->name('something')
            ->schema([
                TextEntry::make('name'),
                Actions::make([
                    Action::make('short list')
                          ->form([
                          ])
                          ->action(function (array $data): void {
                          })
                          ->slideOver(),
                ]),
            ]);
    }
    ```
    
    Let me know what I am doing wrong here

**Comment by danharrin** [comment]
Fixed by #8396. **Please note that infolist action modals do not currently support being nested within other action modals, there is too much technical complexity involved to be maintainable right now.**
